sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -254,7 +262,27 @@ public ChatThreadAsyncClient getChatThreadClient(String chatThreadId) {
      */
     CompletableFuture<Response<Void>> deleteChatThread(String chatThreadId, Context context) {
         context = context == null ? Context.NONE : context;
-        return this.chatClient.deleteChatThreadWithResponseAsync(chatThreadId, context);
+        return this.chatClient.deleteChatThreadWithResponseAsync(chatThreadId, context)
+            .exceptionally(throwable -> translateException(throwable));
+    }
+
+    private <T> T translateException(Throwable throwable) {
```

Comment:
T is never used.

Reply:
I use a template here because return type could be Response<CreateChatThreadResult>, PagedResponse<ChatThreadItem> or Response<Void>. Although in this function it's thrown only and there won't be anything returned, I am trying to find a solution to have one function to fit all those types. Do you have any idea for this?

Reply:
It'll be better if we can create a unit test for this chunk of code to fully understand the behaviour. From the documentation, it is not very clear to me whether we should throw or not.

In createChatThread case, there is another thenApply after exceptionally. If you return null here, will the logic inside thenApply being triggered with Completable<Null>? If yes, there will be a null pointer exception. To avoid that, we need to throw, IMHO.

Reply:
If we should throw, this function becomes 'Throwable translate(Throwable t)'. At the place it is used, 'exceptionally(throwable -> { throw translateException(throwable); });

Reply:
Good idea! I'll update.